programming-languages/README.md

```diff
@@ -24,4 +24,29 @@
 ### Tools
 
 * [Compiler Explorer](https://godbolt.org/): Online C++ compiler. It lets you compile small code snippets and look at the generated assembly code. It supports different compilers across various architectures ranging from x86 to Arduinos.
-* [JetBrains CLion](https://www.jetbrains.com/clion/): One of the best C++ IDEs. CLion works on CMake making it easy to use with roscpp. 
\ No newline at end of file
+* [JetBrains CLion](https://www.jetbrains.com/clion/): One of the best C++ IDEs. CLion works on CMake making it easy to use with roscpp. 
+
+
+## Python
+A high level, interpreted, dead-simple, reads-like-english programming language used for a multitude of tasks, from Machine Learning/Data Analysis to deploying Full-Stack Websites.
+
+The [official website](https://www.python.org/) offers further resources, including talks, podcasts, conferences and the like.
+
+### Tutorials
+
+* [Sentdex Python Tutorial Series](https://www.youtube.com/playlist?list=PLQVvvaa0QuDe8XSftW-RAxdo6OmaeL85M): A collection of short videos that deal with everything from installing python, through loops, control flow, functions, upto usage of the package manager pip. Beyond this, the tutorial deals with specific use cases which may be skipped. It is highly recommended that you code along with the videos.
+
+
+### IDEs
+
+* [JetBrains' PyCharm](https://www.jetbrains.com/pycharm/): Fully featured Python IDE, offering powerful autocomplete, virtual environment support, comprehensive debugging, and notebook support. Users of other JetBrains IDEs will feel right at home.
+* [Jupyter Notebooks/ JupyterLab](https://jupyter.org/): Code in true interactive fashion, as you execute blocks and see outputs of unfinished code, which will help you code further. Runs in a browser, and a pip install is all that is required to get it.
+* [vscode](https://code.visualstudio.com/): A lightweight, extensible IDE with support for autocomplete, error-checking,git, ssh support and ipynb notebooks. It has a very powerful debugger and a long list of useful extensions/plugins. This is the favourite IDE of most of the developers and data scientists. 
+
+### Tools
+
+* **Environment/Dependency management**: [Conda](https://docs.conda.io/) or [venv](https://docs.python.org/3/library/venv.html) are typically used to isolate multiple python environments, so you don't end up with conflicting dependencies for different projects. This is highly recommended. It may seem like an inconvinience at first, but will save you a lot of time later on.
+
+### Notable Libraries:
+* [Numpy](https://numpy.org/): Handles all sorts of Linear Algebra, and it's data-types are used in most other mathematical libraries.
+* [Matplotlib](https://matplotlib.org/): Plots and Visualizes all of your data.
```

Comment:
[Pandas](https://pandas.pydata.org/): Can be used for handling data files such as .csv, .xlsx. Has deep integration with Numpy and Matplotlib

Reply:
Also, change Numpy to NumPy

Reply:
Do you think we should add inbuilt handy libraries like `itertools`, `Collections`,etc? These are very useful to optimize codes and write elegant pythonic codes

Reply:
Yes. Add them as well under inbuilt category.